### PR TITLE
fix helm dep build/update doesn't inherit --insecure-skip-tls-verify from helm repo add

### LINKED ISSUE #8868: Helm dep build/update doesn't inherit --insecure-skip-tls-verify from helm repo add
Output of `helm version`:
3.3.4

When using "helm repo add" with the --insecure-skip-tls-verify flag it works as expected. However, when trying to then helm dep build with a Chart.yaml that references that repository, it throws the x509: certificate signed by unknown authority error and doesn't carry over the flag. Helm dep build/update doesn't actually have that flag as a separate option either

### PR DESCRIPTION
**Pull request by yxxhero**
fix helm dep build/update doesn't inherit --insecure-skip-tls-verify from helm repo add

**What this PR does / why we need it**:
resolve https://github.com/helm/helm/issues/8868
**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by yxxhero** [comment]
Because the time is too long, it is too different from the master branch, so I close https://github.com/helm/helm/pull/8896.

**Comment by yxxhero** [comment]
@bacongobbler Thanks very much.

**Comment by bacongobbler** [comment]
> Because the time is too long, it is too different from the master branch, so I close #8896.

In the future, you can `git rebase master` to bring your branch up to date. There should not be a need to close and re-open a new PR.

https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Comment by yxxhero** [comment]
> > Because the time is too long, it is too different from the master branch, so I close #8896.
> 
> In the future, you can `git rebase master` to bring your branch up to date. There should not be a need to close and re-open a new PR.
> 
> https://git-scm.com/book/en/v2/Git-Branching-Rebasing

Thanks very much， Thank you for your guidance.

**Comment by yxxhero** [comment]
@bacongobbler Looking forward to your guidance!

**Comment by yxxhero** [comment]
Auy update？

**Comment by yxxhero** [comment]
ping

**Comment by yxxhero** [comment]
@mattfarina ping

File: pkg/downloader/manager_test.go [hunk 1]
```diff
@@ -77,15 +77,15 @@ func TestFindChartURL(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	name := "alpine"
-	version := "0.1.0"
-	repoURL := "http://example.com/charts"
```
Comment (bacongobbler) [inline pkg/downloader/manager_test.go:82]: Can you please write a new test case alongside the existing test case?

It's important that existing test cases are left intact and are not re-written. There could be other code paths this test case was intended to test for, and by rewriting it you could potentially break that old test.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Review by bacongobbler — COMMENTED** [review]
The code in this package is getting a bit ridiculous. It's incredibly difficult to read through this code... It's ripe for a refactor at some point.

Regardless, this seems fine. Please reinstate the existing test cases and I'll approve the PR.

**Comment by yxxhero** [comment]
Thanks you very much. This code is very long time. I will change it.  Thank your guidance very much again! @bacongobbler

**Comment by yxxhero** [comment]
I change the unittest, The errors in `ci/circleci: build` I thank is not my fault.   Looking forward to your guidance! @bacongobbler

**Comment by yxxhero** [comment]
@bacongobbler I want to join in `The Helm Project` very much , what should I do next? Thanks very much. I think I can do better.

**Comment by yxxhero** [comment]
The tests is all ok. @bacongobbler  Thank you very much.

**Comment by yxxhero** [comment]
@bacongobbler Looking forward to your guidance! Thanks very much.

**Comment by tculp** [comment]
Any update on this? A fix for this bug would be very useful.

**Comment by yxxhero** [comment]
@bacongobbler What should I do next?

File: pkg/downloader/manager_test.go (hunk 1 above)
Comment (yxxhero) [reply pkg/downloader/manager_test.go:82]: fix.

**Review by bacongobbler — APPROVED** [review]
LGTM. Needs another maintainer to approve before we can merge as per the contributing policy. Thanks.

**Comment by yxxhero** [comment]
@bacongobbler Thanks very much!

**Comment by yxxhero** [comment]
@jdolitsky Can you approve this PR? Thanks very much.

**Comment by yxxhero** [comment]
any update?

**Comment by kofl91** [comment]
I build this locally and am using it instead of the originial release until merged. Thanks @yxxhero for this contribution.

**Comment by yxxhero** [comment]
@kofl91 Thanks very much.

**Comment by yxxhero** [comment]
@mattfarina  can you apporved this. Thanks very much.

**Review by kofl91 — APPROVED** [review]

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @yxxhero

**Pull request merged**